component/src/test/java/org/wso2/extension/siddhi/io/nats/sink/NATSSinkTestCase.java

```diff
@@ -0,0 +1,343 @@
+        simpleWebServer.start();
+        port = simpleWebServer.getMappedPort(4222);
+        Thread.sleep(500);
+    }
+
+    /**
+     * Test for configure the NATS Sink to publish the message to a NATS-streaming subject.
+     */
+    @Test
+    public void natsSimplePublishTest() throws InterruptedException, TimeoutException, IOException {
+        ResultContainer resultContainer = new ResultContainer(2,3);
+        NATSClient NATSClient = new NATSClient("test-cluster","stan_test1","nats://localhost:"
+                + port, resultContainer);
+        NATSClient.connect();
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "@App:name('Test-plan1')\n"
+                + "@sink(type='nats', @map(type='xml'), "
+                + "destination='nats-test1', "
+                + "client.id='test-plan1-siddhi',"
+                + "bootstrap.servers='" + "nats://localhost:"+ port +"', "
+                + "cluster.id='test-cluster'"
+                + ")"
+                + "define stream inputStream (name string, age int, country string);";
+
+        NATSClient.subsripeFromNow("nats-test1");
+        SiddhiAppRuntime executionPlanRuntime = siddhiManager.
+                createSiddhiAppRuntime(inStreamDefinition);
+        InputHandler inputStream = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+
+        inputStream.send(new Object[]{"JAMES", 23, "USA"});
+        inputStream.send(new Object[]{"MIKE", 23, "Germany"});
+        Thread.sleep(1000);
+        Assert.assertTrue(resultContainer.assertMessageContent("JAMES"));
+        Assert.assertTrue(resultContainer.assertMessageContent("MIKE"));
+
+        siddhiManager.shutdown();
+    }
+
+    /**
+     * if a property missing from the siddhi stan sink which defined as mandatory in the extension definition, then
+     * {@link SiddhiAppValidationException} will be thrown.
+     */
+    @Test
+    public void testMissingNatsMandatoryProperty(){
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String inStreamDefinition = "@App:name('Test-plan2')\n"
+                + "@sink(type='nats', @map(type='xml'), "
+                + "bootstrap.servers='" + "nats://localhost:"+ port +"', "
+                + "client.id='test-plan2',"
+                + "cluster.id='test-cluster'"
+                + ")"
+                + "define stream inputStream (name string, age int, country string);";
+
+        try {
+            SiddhiAppRuntime executionPlanRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition);
+            Assert.fail();
+        } catch (SiddhiAppValidationException e) {
+            Assert.assertTrue(e.getMessage().contains("Option 'destination' does not exist in the configuration of "
+                    + "'sink:nats'"));
+        }
+    }
+
+    /**
+     * If invalid NATS url provided then {@link SiddhiAppCreationException} will be thrown
+     */
+    @Test
```

Comment:
Assert using @Test(expectedException=.......)

Reply:
Resolved with the commit 77f7e16ab934662a2382acbf160ccb0448d47bb4.